### ISSUE TITLE
Adaptive bitrate wildcard

### ISSUE THREAD

**Issue by i2iSteve** [kaltura/nginx-vod-module#1370]
We are currently testing the VOD module and are very impressed with it. Though we have run into one issue:

The files are located on s3, we have tested it and it is running fine on 1000k and 2000k, our issue is when it comes to a file that doesn't have a bitrate encoded already the urlmap fails with a 404 and wont play the video. Is there any way around this that doesn't involve a mapping file?

**Comment by erankor** [comment]
Sorry, I don't understand the question... what is the error you get?

**Comment by i2iSteve** [comment]
Sorry, for example we have:

test_1000k.mp4
test_2000k,mp4
test2_1000k.mp4

If we use 1000,2000,k.mp4 test_ works but test2 will fail as test2_2000k.mp4 is missing. Due to the large number of files we cannot reencode them all. Also json wouldn't be a effective way forward. Though if all else fails, we can code it in laravel.

**Comment by erankor** [comment]
I don't think you need to reencode anything, but the service that builds the playback URL needs to know what bitrates actually exist in S3. According to that, it can decide whether to add 2000k to the URL or 4000k or whatever...
